### PR TITLE
Mixtral enablement. 

### PR DESCRIPTION
**Pull request by wang2yn84**
Mixtral 8x7b model is working for both offline and online, bf16 and int8. Let's get this in first so we can parallelize the work. Will add tests in the coming PRs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,12 +64,21 @@ huggingface-cli download google/gemma-7b-pytorch --local-dir $input_ckpt_dir
...
+huggingface-cli download mistralai/Mixtral-8x7B-v0.1 --local-dir $input_ckpt_dir
+```
+
 ## Run weight safetensor convert
 
 ```bash
 export input_ckpt_dir=Original llama weights directory
 export output_ckpt_dir=The output directory
-export model_name="llama-3" # or "llama-2", "gemma"
+export model_name="llama-3" # or "llama-2", "gemma", "mistral"
```
Comment (qihqi) [inline README.md:81]: change this to mixtral

File: jetstream_pt/third_party/mistral/model_original.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        torch.empty(config.num_experts, config.intermediate_size, config.dim)
+    )
+    self.w2 = nn.Parameter(
+        torch.empty(config.num_experts, config.dim, config.intermediate_size)
+    )
+    self.w3 = nn.Parameter(
+        torch.empty(config.num_experts, config.intermediate_size, config.dim)
+    )
+
+  def forward(self, x: Tensor, expert_indices: Tensor) -> Tensor:
```
Comment (qihqi) [inline jetstream_pt/third_party/mistral/model_original.py:204]: I had a change to use different logic for longer seqlen and i pushed to your branch, is that lost from merging?

Comment (qihqi) [reply jetstream_pt/third_party/mistral/model_original.py:204]: also the quantized change

**Comment by qihqi** [comment]
please make sure the name is mixtral and not mistral. We might add mistral 7b ( the non-Moe version) later, so it would be confusing

File: jetstream_pt/third_party/mistral/model_original.py (hunk 2 above)
Comment (wang2yn84) [reply jetstream_pt/third_party/mistral/model_original.py:204]: This is the original model. Your changes are in model.py

File: README.md (hunk 1 above)
Comment (wang2yn84) [reply README.md]: Thanks. I was confused about the name initially and that's why there are mixes of mistral and mixtral. I also changes everything to Mixtral. Done.

File: convert_checkpoints.py [hunk 3]
```diff
@@ -462,6 +471,99 @@ def _get_gemma_state_dict(input_ckpt_dir):
...
+
+  config = json.loads((input_ckpt_dir / "config.json").read_text())
+  print(f"Loaded config: {config}")
+  weight_map = {
+      "tok_embeddings.weight": "tok_embeddings.weight",
+      "layers.{}.attention.wq.weight": "layers.{}.attention.wq.weight",
+      "layers.{}.attention.wk.weight": "layers.{}.attention.wk.weight",
+      "layers.{}.attention.wv.weight": "layers.{}.attention.wv.weight",
+      "layers.{}.attention.wo.weight": "layers.{}.attention.wo.weight",
+      "layers.{}.block_sparse_moe.w1": "layers.{}.block_sparse_moe.cond_ffn.w1",
```
Comment (FanhaiLu1) [inline convert_checkpoints.py:501]: Looks like only these weight name are difference, can we only store the the different name in the map?

**Review by FanhaiLu1 — COMMENTED** [review]
Thanks for add Mixtral, the code is clean and overlay look good!

File: convert_checkpoints.py (hunk 3 above)
Comment (wang2yn84) [reply convert_checkpoints.py:496]: Good point, removed

**Review by qihqi — APPROVED** [review]

**Review by FanhaiLu1 — APPROVED** [review]

**Pull request merged**